libkmod/libkmod-signature.c

```diff
@@ -321,21 +321,21 @@ bool kmod_module_signature_info(const struct kmod_file *file, struct kmod_signat
 	if (size < (off_t)sizeof(struct module_signature))
 		return false;
 	size -= sizeof(struct module_signature);
-	modsig = (struct module_signature *)(mem + size);
-	if (modsig->algo >= PKEY_ALGO__LAST ||
-			modsig->hash >= PKEY_HASH__LAST ||
-			modsig->id_type >= PKEY_ID_TYPE__LAST)
+	memcpy(&modsig, mem + size, sizeof(struct module_signature));
+	if (modsig.algo >= PKEY_ALGO__LAST ||
+			modsig.hash >= PKEY_HASH__LAST ||
+			modsig.id_type >= PKEY_ID_TYPE__LAST)
 		return false;
-	sig_len = be32toh(get_unaligned(&modsig->sig_len));
+	sig_len = be32toh(get_unaligned(&modsig.sig_len));
```

Comment:
this get_unaligned() here doesn't make sense anymore after you copied the struct.

Reply:
on a second thought this looks more a false positive. The alignment it's complaining is for the pointer... but it doesn't make much sense. The only field where the alignment matters is already taken care by using get_unaligned().

Reply:
It's a funny false positive.

I've debugged modinfo on my x86_64 system and in fact, with default compiler options and at least gcc 14.2.0 the unaligned access code is optimized away. That's a proper optimization, since x86_64 has no problem with unaligned access.

But ASAN checks the optimized code and warns that unaligned access occurs. Which is objectively correct. It's just not an issue on my system.

So yes, I drop this PR because everything works as intended. I wondered anyway why this was never spotted by anyone over around 10 years. These rather uncommon architectures cannot be THAT uncommon.

Reply:
I'm not really against applying it if it allows us to enable the sanitizer in future. With the right commit message, we can apply it.